calling-convention.adoc

```diff
@@ -2,11 +2,19 @@
 == Calling Convention
 
 In the RISC-V psABI, the vector registers `v0`-`v31` are all caller-saved.
-The `vstart`, `vl`, and `vtype` CSRs are also caller-saved.
+The `vl` and `vtype` CSRs are also caller-saved.
 
-The `vxrm` and `vxsat` fields have thread storage duration.
+Procedures may assume that `vstart` is zero upon entry.  Procedures may
+assume that `vstart` is zero upon return from a procedure call.
 
-Executing a system call causes `v0`-`v31` to become unspecified.
+NOTE: Application software should normally not write `vstart` explicitly.
+Any procedure that does explicitly write `vstart` to a nonzero value must
+zero `vstart` before either returning or calling another procedure.
+
+The `vxrm` and `vxsat` fields of `vcsr` have thread storage duration.
+
+Executing a system call causes all caller-saved vector registers
+(`v0`-`v31`, `vl`, `vtype`) and `vstart` to become unspecified
```

Comment:
The clarifying note should also be revised:
```
The values that v0-v31 assume after a system call cannot expose information from 
other processes, so typically the registers will either remain intact or will be zeroed.
```
As this could easily be misconstrued as an OS requirement to be secure.
Rather some wording with this intent:

After a system call the values in v0-v31 will typically 
either remain intact or will be zeroed so that the OS does not expose information from other processes.

or

So that the OS does not expose information from other processes., the values in v0-v31 will typically 
either remain intact or be zeroed after a system call.

Reply:
Ack.